core/trino-main/src/main/java/io/trino/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -175,12 +176,13 @@ private AggregationAnalyzer(
                 .map(expression -> scopeAwareKey(expression, analysis, sourceScope))
                 .collect(toImmutableSet());
 
-        this.columnReferences = analysis.getColumnReferenceFields()
-                .entrySet().stream()
-                .collect(toImmutableMap(Map.Entry::getKey, entry -> entry.getValue().getFieldId()));
+        Map<Expression, FieldId> columnReferencesLocal = new IdentityHashMap<>();
```

Comment:
I kind of feel that this requires a comment maybe? To justify why using identity is ok in this context and why it has been used.